File: src-tauri/src/domain/response/transfer.rs
use crate::model::transfer::{
    TransferMonthAmountResponse, TransferResponse, TransferResponseDeleteAt,
    TransferResponseMonthStatusFailed, TransferResponseMonthStatusSuccess,
    TransferResponseYearStatusFailed, TransferResponseYearStatusSuccess,
    TransferYearAmountResponse,
};

use super::PaginationMeta;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransferMonthStatusSuccess {
    pub status: String,
    pub message: String,
    pub data: Vec<TransferResponseMonthStatusSuccess>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransferYearStatusSuccess {
    pub status: String,
    pub message: String,
    pub data: Vec<TransferResponseYearStatusSuccess>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransferMonthStatusFailed {
    pub status: String,
    pub message: String,
    pub data: Vec<TransferResponseMonthStatusFailed>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransferYearStatusFailed {
    pub status: String,
    pub message: String,
    pub data: Vec<TransferResponseYearStatusFailed>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransferMonthAmount {
    pub status: String,
    pub message: String,
    pub data: Vec<TransferMonthAmountResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransferYearAmount {
    pub status: String,
    pub message: String,
    pub data: Vec<TransferYearAmountResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransfer {
    pub status: String,
    pub message: String,
    pub data: Option<TransferResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransfers {
    pub status: String,
    pub message: String,
    pub data: Vec<TransferResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransferDelete {
    pub status: String,
    pub message: String,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransferAll {
    pub status: String,
    pub message: String,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponsePaginationTransfer {
    pub status: String,
    pub message: String,
    pub data: Vec<TransferResponse>,
    pub pagination: Option<PaginationMeta>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponsePaginationTransferDeleteAt {
    pub status: String,
    pub message: String,
    pub data: Vec<TransferResponseDeleteAt>,
    pub pagination: Option<PaginationMeta>,
}

File: src/types/domain/response/transfer/transfer.ts
import {
  Transfer,
  TransferMonthAmount,
  TransferMonthStatusFailed,
  TransferMonthStatusSuccess,
  TransferTrashed,
  TransferYearAmount,
  TransferYearStatusFailed,
  TransferYearStatusSuccess,
} from "@/types/model";
import { PaginationMeta } from "../pagination";

export interface ApiResponseTransferMonthStatusSuccess {
  status: string;
  message: string;
  data: TransferMonthStatusSuccess[];
}

export interface ApiResponseTransferYearStatusSuccess {
  status: string;
  message: string;
  data: TransferYearStatusSuccess[];
}

export interface ApiResponseTransferMonthStatusFailed {
  status: string;
  message: string;
  data: TransferMonthStatusFailed[];
}

export interface ApiResponseTransferYearStatusFailed {
  status: string;
  message: string;
  data: TransferYearStatusFailed[];
}

export interface ApiResponseTransferMonthAmount {
  status: string;
  message: string;
  data: TransferMonthAmount[];
}

export interface ApiResponseTransferYearAmount {
  status: string;
  message: string;
  data: TransferYearAmount[];
}

export interface ApiResponseTransfer {
  status: string;
  message: string;
  data?: Transfer;
}

export interface ApiResponseTransfers {
  status: string;
  message: string;
  data: Transfer[];
}

export interface ApiResponseTransferDelete {
  status: string;
  message: string;
}

export interface ApiResponseTransferAll {
  status: string;
  message: string;
}

export interface ApiResponsePaginationTransfer {
  status: string;
  message: string;
  data: Transfer[];
  pagination: PaginationMeta;
}

export interface ApiResponsePaginationTransferDeleteAt {
  status: string;
  message: string;
  data: TransferTrashed[];
  pagination: PaginationMeta;
}
